File: LeetCode/1-100/53. 最大子数组和.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var maxSubArray = function (nums) {
  let prev = 0,
    max = nums[0]
  nums.forEach((v) => {
    prev = Math.max(prev + v, v)
    console.log(prev, 'prev')
    max = Math.max(max, prev)
    console.log(max, 'max')
  })
  return max
}

File: LeetCode/1-100/53. 最大子数组和.java
class Solution {
    public int maxSubArray(int[] nums) {
         int prevMax = 0,max = nums[0];
        for (int num : nums) {
            prevMax = Math.max(num,prevMax + num);
            max = Math.max(max,prevMax);
        }
        return max;
    }
}